examples/otlp/grpc_metric_main.cc

```diff
@@ -0,0 +1,98 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ENABLE_METRICS_PREVIEW
+
+#  include "opentelemetry/exporters/otlp/otlp_grpc_metric_exporter.h"
+#  include "opentelemetry/metrics/provider.h"
+#  include "opentelemetry/sdk/metrics/aggregation/default_aggregation.h"
+#  include "opentelemetry/sdk/metrics/export/periodic_exporting_metric_reader.h"
+#  include "opentelemetry/sdk/metrics/meter.h"
+#  include "opentelemetry/sdk/metrics/meter_provider.h"
+
+#  include <memory>
+#  include <thread>
+
+#  ifdef BAZEL_BUILD
+#    include "examples/common/metrics_foo_library/foo_library.h"
+#  else
+#    include "metrics_foo_library/foo_library.h"
+#  endif
+
+namespace metric_sdk    = opentelemetry::sdk::metrics;
+namespace nostd         = opentelemetry::nostd;
+namespace common        = opentelemetry::common;
+namespace metrics_api   = opentelemetry::metrics;
+namespace otlp_exporter = opentelemetry::exporter::otlp;
+
+namespace
+{
+
+otlp_exporter::OtlpGrpcMetricExporterOptions options;
+
+void initMetrics()
+{
+  std::unique_ptr<metric_sdk::MetricExporter> exporter{
+      new otlp_exporter::OtlpGrpcMetricExporter(options)};
```

Comment:
Could we use `OtlpGrpcMetricExporterFactory` after #1606  be merged? Just like other examples?
It think we also need factories for metrics SDKs, though this is not a problem of this PR.

Reply:
Yes, using OtlpGrpcMetricExporterFactory now.  For factories for metrics SDKs, we have #1485. This is supposed to be done once SDK is nearing GA.